app/Routes/Route.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Routes;
+
+use App\Controllers\ControllerInterface;
+
+final readonly class Route
+{
+    public function __construct(
+        private string $method,
+        private string $route,
+        private ControllerInterface $controller,
+        private string $action
+    ) {
+    }
+
+    public function getActionCallback(): callable
+    {
+        return [$this->controller, $this->action];
+    }
+
+    public function matchesWithParameters(string $method, string $uri): bool
```

Comment:
There are no parameters in method's parameters 😄 

just leave simple short name `matches`